README.md

```diff
@@ -0,0 +1,412 @@
+{
+}
+```
+
+There should be no space around member operators `.` and `->`.
+
+Address-of `&` and de-reference `*` unary operators should be always adjacent to the variable they affect.
+
+When overloading an operator, there should be a space between the `operator` keyword and the operator being overloaded.
+
+```c++
+Subscriber& Subscriber::operator <<(
+        const qos::SubscriberQos& qos)
+{
+    this->qos(qos);
+    return *this;
+}
+```
+
+#### Templates
+
+There should be a space between closing angles `>`.
+
+```c++
+std::map<std::string, std::vector<DataWriterImpl*> > writers_;
+```
+
+#### Line-end Comments
+
+Lines that are non-obvious should get a comment at the end of the line.
+These end-of-line comments should be separated from the code by a space.
+
+### Function Declarations and Definitions
+
+To improve parameter readability, each parameter must be in its own line, even the first one, and even if there is only one parameter.
+The line-wrapping indentation is to be used in this case.
+Never indent to the opening parenthesis level.
+
+This does not apply when calling the function.
+In this case the parameters should be in the same line taking into account the space after each comma that separates the parameters.
+
+The closing parenthesis `)` should be adjacent to the last parameter on the list.
+
+The opening and close braces should be each alone in their own line.
+
+Method definition should not have any indentation.
+
+```c++
+void EDPStatic::assignRemoteEndpoints(
+        const ParticipantProxyData& pdata)
+{
+    // Method body
+}
+                                        // Blank line between methods
+// Incorrect
+void EDPStatic::assignRemoteEndpoints(const ParticipantProxyData& pdata)
+{
+    // Method body
+}
+
+bool EDPStatic::newRemoteReader(
+        const GUID_t& participant_guid,
+        const string_255& participant_name,
+        uint16_t user_id,
+        EntityId_t ent_id)
+{
+    // Method body
+}
+```
+
+### Loops and Conditionals
+
+There must always be braces `{ }`, even if there is no need (one line statements or even empty statements).
+Additionally, both the opening and closing braces should be each alone in their own new line.
+
+```c++
+// Correct
+if (!error)
+{
+    return success;
+}
+
+// Incorrect
+if (!error)
+    return success;
+
+// Incorrect
+if (!error) return success;
+```
+
+### Pointer and Reference Expressions
+
+When declaring a pointer or reference variable, the `*` or `&` must be always adjacent to the type and separated from the variable name by a space.
+There is an exception when defining several pointers or references in the same line. In this case the `*` or `&` must be adjacent to the name. 
+This also applies when declaring a prototype and defining a function or method.
+
+In the case of defining double pointers (pointer to pointer), there should be no space between the asterisks `*`.
+
+```c++
+char* p;
+int** ptr;
+int& x;
+
+int *value, *key;
```

Comment:
No exceptions!

Reply:
I agree with @LuisGP here. I think we should add a rule saying that we don't allow several declarations on the same line.

Reply:
The rule has been included.